### PR TITLE
Task/284/concatenate all pipelines

### PR DESCRIPTION
**Pull request by cyberbikepunk**
@akariv At this stage the resulting merged dataset doesn't pass the validation tests. I about to figure out why. So, let's wait until I fix that to merge this PR.

Also, I'm not quite sure how to go about testing the `concatenate` function. I could test it against dummy zip-files, or let it run against the actual `data` directory and test whether I get a generator of dictionaries. The `describe` function is more straightforward. 

I also need to content the Lama and fix a couple of source files to make Travis happy.

### CODE REVIEW

**Comment by cyberbikepunk** [comment]
And i just realized i forgot to cherry pick the first commit and put it in another PR. It has to do with the flat source file.

**Comment by cyberbikepunk** [comment]
@akariv This PR ends up being a bit of a mixed bag because I fixed minor unrelated stuff along the way. The concatenation pipeline is defined [here](https://github.com/os-data/eu-structural-funds/blob/e0d592d6ef438c2350ad9c35271f00df980f02ec/pipeline-spec.yaml). The main file to look at is the ['concatenate_all_pipelines'](https://github.com/os-data/eu-structural-funds/blob/ae39f5f742d0423e437b494aea84ba44b8f82203/common/processors/concatenate_all_pipelines.py) processor. The local file collection and the resource iteration are now separate functions. I also added some [unit-tests](https://github.com/os-data/eu-structural-funds/blob/ae39f5f742d0423e437b494aea84ba44b8f82203/tests/processors/test_concatenate_all_pipelines.py).

There is still a `charmap` failure reported during the upload. I need to figure out which file causes the problem.

**Comment by cyberbikepunk** [comment]
@akariv Travis is not happy because `pandas` is required twice. I saw that you added it to the `requirement.txt`. It was already in the `requirements.dev.txt` file. That was on purpose because it's not required for the pipeline to run, only for producing stats form the `bootstrap` command line. Funny that only 3.3 and 3.4 versions complain. What's the best practice about this?

**Comment by akariv** [comment]
On the server I only install what's in `requirements.txt`. I added it because it was needed for running `bootstrap update` (which is a prerequisite for running the pipelines). 
If it makes no difference to you, then let's remove it from the dev file.

File: common/processors/concatenate_all_pipelines.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    """Return a single resource generator for all datasets."""
+
+    nb_files = 0
+
+    for filename, csv_text in csv_files:
+        nb_files += 1
+
+        with Stream(csv_text, **STREAM_OPTIONS) as stream:
+            for row in stream.iter(keyed=True):
+                yield row
```
Comment (akariv) [inline common/processors/concatenate_all_pipelines.py]: Didn't this concatenated file was supposed to contain only the 5 common columns?

Comment (cyberbikepunk) [reply common/processors/concatenate_all_pipelines.py]: @akariv you have a point. My approach was to let all columns through and then do QA on the minimal set of columns. But I like your idea of putting a column throttle here. I'll add a parameter to the processor to select a subset of columns.

**Comment by akariv** [comment]
OK, LGTM

**Comment by cyberbikepunk** [comment]
This PR solves #291 .

**Pull request merged**